File: fixtures/scale/julia/small/src/Wrap.jl
module Wrap

export wrap_text

function wrap_line(line::String, width::Int)
    chunks = String[]
    while length(line) > width
        push!(chunks, line[1:width])
        line = line[(width + 1):end]
    end
    push!(chunks, line)
    return chunks
end

function wrap_text(lines::Vector{String}, width::Int)
    return reduce(vcat, [wrap_line(line, width) for line in lines])
end

end

File: fixtures/scale/r/small/lib/wrap.r
# Fixed-width line wrapping.

wrap_line <- function(line, width) {
  chunks <- character(0)
  while (nchar(line) > width) {
    chunks <- c(chunks, substr(line, 1, width))
    line <- substr(line, width + 1, nchar(line))
  }
  c(chunks, line)
}

wrap_text <- function(lines, width) {
  unlist(lapply(lines, wrap_line, width = width))
}
